torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1269,3 +1289,39 @@ def _optim_state_dict(
         )
 
     return fsdp_osd
+
+
+def _get_fqn_to_fsdp_param_info(model: nn.Module) -> Dict[str, FSDPParamInfo]:
+    """
+    Construct the maaping from a param's fqn to its corresponding FSDPParamInfo
+    if the param is managed by FSDP. FlatParam only stores the first FQN of a
+    shared parameter. So the keys in the mapping are guranteed to map to unique
+    parameters.
+    """
+    def module_fn(module, prefix, fqn_to_param_info):
```

Comment:
Should we add a comment saying we need to use `_apply_to_modules` to get the global FQN (since the saved FQNs are like local FQNs, not necessarily prefixed from the global root module)?